### PR TITLE
Support for Python 3.12

### PR DESCRIPTION
**Pull request by dmichaels-harvard**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -143,3 +143,34 @@ jobs:
       - name: QA (unit)
         run: |
           poetry run invoke test --no-flake
+
+  build_3_12:
```
Comment (willronchetti) [inline .github/workflows/main.yml:147]: I've just realized this build is not defined correctly - you can re-use all the below by using the `matrix` syntax, see https://github.com/4dn-dcic/utils/blob/master/.github/workflows/main.yml#L24-L26 as an example

**Review by willronchetti — APPROVED** [review]
Alex V should deploy/test this out before merging

**Comment by alexander-veit** [comment]
I am getting `SyntaxWarnings` at various places when using Python 3.12, e.g., 
```
/home/ubuntu/GitHub/tibanna/tibanna/awsem.py:396: SyntaxWarning: invalid escape sequence '\.'
  filename = re.match('(.+)\.{0}$'.format(unzip), filename).group(1)
```
This is exptected with Python 3.12: https://docs.python.org/dev/whatsnew/3.12.html#other-language-changes
I think we need to use raw string instead.

**Review by alexander-veit — APPROVED** [review]
I deployed it and ran a test workflow and everything seems to work for Python 3.12. I would just modify the regex expressions to avoid the syntax warnings, e.g., replace 
```
filename = re.match('(.+)\.{0}$'.format(unzip), filename).group(1)
```
with 
```
filename = re.match(r'(.+)\.{0}$'.format(unzip), filename).group(1)
```

**Pull request merged**